src/App.jsx

```diff
@@ -1,35 +1,79 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useState, useEffect } from "react";
+import axios from "axios";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [articles, setArticles] = useState([]);
+  const [selectedArticle, setSelectedArticle] = useState(null);
+
+  useEffect(() => {
+    axios.get("https://nc-news-rppe.onrender.com/api/articles").then((res) => {
+      setArticles(res.data.articles);
+    });
+  }, []);
+
+  const buttonHandler = (articleId) => {
+    // console.log("Button clicked for article ID:", articleId);
+    const selected = articles.find(
+      (article) => article.article_id === articleId
+    );
+    // console.log("Selected article:", selected);
+    setSelectedArticle(selected);
+  };
 
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
+      <div className="container my-5">
+        <h1>NC-NEWS APP</h1>
+
+        <div className="row text-center">
+          {articles.map((article, index) => {
+            return (
+              <div className="col" key={index} style={{ marginBottom: "20px" }}>
```

Comment:
you could separate this out into another component, I can see you have ArticleItem.jsx, this might be better in there